voter-dapp/src/RetrieveRewards.js

```diff
@@ -14,13 +15,13 @@ function getOrCreateObj(containingObj, field) {
   return containingObj[field];
 }
 
-function useRetrieveRewardsTxn(retrievedRewardsEvents, revealedVoteEvents, priceResolvedEvents, votingAccount) {
+function useRetrieveRewardsTxn(retrievedRewardsEvents, revealedVoteEvents, priceRequests, votingAccount) {
   const { drizzle, useCacheSend } = drizzleReactHooks.useDrizzle();
   const { web3 } = drizzle;
 
   const { send, status } = useCacheSend("Voting", "retrieveRewards");
 
-  if (retrievedRewardsEvents === undefined || revealedVoteEvents === undefined || priceResolvedEvents === undefined) {
+  if (retrievedRewardsEvents === undefined || revealedVoteEvents === undefined || !priceRequests) {
```

Comment:
nit: it seems to me all of these checks are doing the same thing (whether some variable is `undefined`) but why do you use `!priceRequests` versus `revealedVoteEvents === undefined`. I think it'd be a bit cleaner to use all "!"'s

Reply:
Yeah, this is a drizzle thing.

The first two come from naked drizzle calls. Naked drizzle calls return `undefined` to mean "result hasn't come back yet" and `null` to mean "error or empty result". The first two should generally never have `null`, so I check for undefined specifically, since I only want to catch "hasn't come back yet". I don't want this to silently ignore real errors or empty values coming back from these calls.

The last one should really only be `null`, since we dictate the return values for that one (not a naked drizzle call). I should probably explicitly check for null, tbh. I'll change that.

Reply:
Note: I changed the last one to null. Do you think that makes sense?

Reply:
@nicholaspai I updated the third object that we control to be undefined if the call hasn't finished resolving yet to make it work like the regular drizzle calls. This allowed me to make them all the same `=== undefined`. PTAL.

Reply:
>The first two come from naked drizzle calls. Naked drizzle calls return undefined to mean "result hasn't come back yet" and null to mean "error or empty result". 

This is horrible lol. I understand though

Reply:
> This is horrible lol. I understand though
lol yup